docs/03_server/04_event-handler/03_advanced.md

```diff
@@ -241,110 +251,247 @@ eventHandler {
 
-With this xml configuration, you can look up the user's rights in the database and return `true` only if the necessary rights are in place. For example, if your system has the concept of internal and external users and you only want to allow internal users to accept pending events, then you could check your custom user "ACCESS_TYPE" field as follows:
+- `selectPredicate` is a function that accepts an indeterminate number of functions returning a boolean value, as well as a mandatory `default` function to handle messages that do not fall into any defined category. The `default` function will provide a [GenesisSet](../../03_server/09_inter-process-messages/02_genesisset.md) object with the contents of the original message payload.
+- `onEvent` works very similarly to any other GPAL [event handler definition](../../server/event-handler/basics.md#adding-a-name). It allows us to treat the incoming message in the same way we would have done it within the original event handler, but each function must return a boolean expression.
 
-```xml
-        ...
-        <acceptPendingApproval>
-            <![CDATA[
-               def searchRecord = new DbRecord("USER_ATTRIBUTES")
-               def userAttributes = db.get(searchRecord, "USER_ATTRIBUTES_BY_USER_NAME")
-               userAttributes?.getString("ACCESS_TYPE") == "INTERNAL"
-            ]]>
-        </acceptPendingApproval>
-        ...
+
+Please see the example below:
+
+```kotlin
+pendingApproval {
+    accept {
+        selectPredicate(        
+            onEvent<TradeInsert>("TRADE_INSERT") { event ->
+                val tradeInsert = event.details
+                // Allows acceptance of pending approval events for EVENT_TRADE_INSERT, only if the trade is for TSLA
+                tradeInsert.symbol == "TSLA"
+            },
+            onEvent<TradeAmend>("TRADE_AMEND") { event ->
+                val tradeAmend = event.details
+                // Allows acceptance of pending approval events for EVENT_TRADE_AMEND, only if the counterparty name for it is "GENESIS"
+                tradeAmend.counterpartyName == "GENESIS"
+            },
+            onEvent<TradeDelete>("TRADE_DELETE") { event ->
+                // Allows acceptance of pending approval events for EVENT_TRADE_DELETE in all scenarios.
+                true
+            },
+            // If the message can't be deserialized we will use a default fallback to genesisSet.
+            default =  { genesisSet ->
+                true
+            }
+        )
+    }
+```
+### System rejects
```

Comment:
Should this start by just saying that because we're just holding onto an event and releasing when it is accepted, if the event fails validation (e.g. data it relies upon has changed since being sent for approval) that it will result in REJECTED_BY_SYSTEM